pkg/networkservice/nsmonitor/client.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package nsmonitor
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/common"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+)
+
+type netNSMonitorClient struct {
+	monitor *netNSMonitor
+}
+
+// NewClient returns new net ns monitoring client
+func NewClient() networkservice.NetworkServiceClient {
```

Comment:
Consider to add a possible to pass a iface to manitor into the client for unit testing.

```suggestion
type Option func(*options) 
type Monitor interface {
SubscribeOnDelete(context.Context, *url.URL)<- chan struct{}
}
func WithMonitor(m Monitor) {
    return func(o *options) {
        o.monitor= m
    }
}
func NewClient(chainContex context.Context, opts...Option) networkservice.NetworkServiceClient {
```